components/ui/form/Schedule.tsx

```diff
@@ -62,6 +65,41 @@ const TimeRangeField = ({ name }: TimeRangeFieldProps) => {
     );
   }, []);
 
+  const [valueSelectStart, setValueStart] = useState<Option>(initialValueforSelectStart);
+
+  const customOnChange = (onChange: any, option: any) => {
+    onChange(new Date(option?.value as number));
+    setValueStart(option);
+  };
+
+  const calculateRange = (value: Option) => {
+    const earlyMorningHours = [{ open: "00:00", close: "04:45" }];
+    let opened = false;
+    earlyMorningHours.forEach((item) => {
+      const open = new Date("1/1/1999 " + item.open);
+      const close = new Date("1/1/1999 " + item.close);
+      const hourDate = new Date(`1/1/1999 ${value.label}`);
+      opened = opened || (hourDate >= open && close >= hourDate);
+    });
+    return opened;
+  };
+
+  const calculateValue = (value: Option) => {
+    if (calculateRange(value)) {
+      if (calculateRange(valueSelectStart)) {
+        if (value.value < valueSelectStart.value) {
+          return valueSelectStart;
+        }
+      }
+      return value;
+    }
+
+    if (value.value < valueSelectStart.value) {
+      return valueSelectStart;
+    }
+    return value;
+  };
```

Comment:
@Jenietoc nesting control flow structures are not a good approach. IMO, you can improve this block of code